lib/Fhp/BaseAction.php

```diff
@@ -165,6 +143,31 @@ public function ensureDone()
      */
     abstract public function createRequest(BPD $bpd, ?UPD $upd);
 
+    /**
+     * Called by FinTs::execute when this action is about to be executed, in order to get a request. This function can
+     * be called multiple times in case the response is paginated.
+     * This method also tries to check if the segments might need a tan and stores this information for use in
+     * FinTs::execute
+     * @param BPD|null $bpd See {@link BPD}.
+     * @param UPD|null $upd See {@link UPD}. This is usually present (non-null), except for a few special login and TAN
+     *     management actions.
+     * @return BaseSegment[] A segment or a series of segments that should be sent to the bank server.
+     *     Note that an action can return an empty array to indicate that it does not need to make a request to the
```

Comment:
Das müssen wir hier nicht nochmal wiederholen, denke ich. Einfach "An empty array means that no request is necessary at all." würde reichen.